core/executor/src/wasm_runtime.rs

```diff
@@ -58,6 +52,16 @@ pub enum WasmExecutionMethod {
 	Compiled,
 }
 
+/// A Wasm runtime object along with its cached runtime version.
+struct VersionedRuntime {
+	runtime: Box<dyn WasmRuntime>,
+	/// Runtime version according to `Core_version`.
+	///
+	/// Can be `None` if the runtime doesn't provide the information or there was an error while
+	/// fetching it.
+	version: Option<RuntimeVersion>,
```

Comment:
The core interface is required by any valid substrate runtime, so there always need to be a version.

If the runtime does not provides a version, it should be handled as invalid.

So, please remove the Option :)